lib/libc/minimal/source/stdout/stdout_console.c

```diff
@@ -22,29 +25,35 @@ void __stdout_hook_install(int (*hook)(int))
 	_stdout_hook = hook;
 }
 
-int fputc(int c, FILE *stream)
+int _impl__zephyr_fputc(int c, FILE *stream)
 {
 	return (stdout == stream) ? _stdout_hook(c) : EOF;
 }
 
+#ifdef CONFIG_USERSPACE
+_SYSCALL_HANDLER(_zephyr_fputc, c, stream)
+{
+	return _impl__zephyr_fputc(c, (FILE *)stream);
+}
+#endif
+
+int fputc(int c, FILE *stream)
+{
+	return _zephyr_fputc(c, stream);
+}
+
 int fputs(const char *_MLIBC_RESTRICT string, FILE *_MLIBC_RESTRICT stream)
 {
-	if (stream != stdout) {
-		return EOF;
-	}
+	int len = strlen(string);
+	int ret;
 
-	while (*string != '\0') {
-		if (_stdout_hook((int)*string) == EOF) {
-			return EOF;
-		}
-		string++;
-	}
+	ret = fwrite(string, len, 1, stream);
```

Comment:
This can call _impl__zephyr_fwrite() directly to avoid going through the `are we in usermode?` branch

Reply:
fputs is not defined as a syscall so I think this needs to be left as-is